### ISSUE TITLE
Add an owned version of WaitForCancellationFuture

### ISSUE THREAD

**Issue by sfackler** [tokio-rs/tokio#4466]
In the vein of `Semaphore::acquire_owned`, it'd be nice if there were an owned variant of `CancellationToken::cancelled`. Since `CancellationToken` is already `Clone`, I think this should be pretty straightforward?

**Comment by Darksonn** [comment]
Hmm. The `acquire_owned` method primarily exists because it returns an owned permit, as opposed to existing so you can get an owned future.

Also, you can already do it today with an async block:
```rust
let owned_future = async move { token_clone.cancelled().await };
```

**Comment by sfackler** [comment]
Yeah, that's what I'm doing now - the main downside is that you can't name the future type without converting it to a `BoxFuture`.

**Comment by Darksonn** [comment]
I would be ok with a PR that adds this.

**Comment by Darksonn** [comment]
Due to #4652, this is no longer easy to implement. I will close this as wont-fix.
